src/audio/codec_adapter/codec_adapter.c

```diff
@@ -530,6 +530,19 @@ static int codec_adapter_trigger(struct comp_dev *dev, int cmd)
 	return comp_set_state(dev, cmd);
 }
 
+static int codec_adapter_reset(struct comp_dev *dev)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+
+	comp_cl_info(&comp_codec_adapter, "codec_adapter_reset(): resetting");
+
+	cd->state = PP_STATE_CREATED;
+
+	comp_cl_info(&comp_codec_adapter, "codec_adapter_reset(): done");
+
+	return comp_set_state(dev, COMP_TRIGGER_RESET);
+}
```

Comment:
I feel this implementation is incomplete; doesn't reset also need to do something to the actual codec?